File: app/src/main/java/com/daniel/mobilepauker2/dropbox/DeleteFileTask.java
package com.daniel.mobilepauker2.dropbox;

import android.os.AsyncTask;

import com.daniel.mobilepauker2.utils.Constants;
import com.dropbox.core.DbxException;
import com.dropbox.core.v2.DbxClientV2;
import com.dropbox.core.v2.files.DeleteResult;

import java.io.File;
import java.util.ArrayList;
import java.util.List;

/**
 * Created by Daniel on 14.03.2018.
 * Masterarbeit:
 * MobilePauker++ - Intuitiv, plattformübergreifend lernen
 * Daniel Fritsch
 * hs-augsburg
 * <p>
 * Löscht die Dateien auf Dropbox.
 */

public class DeleteFileTask extends AsyncTask<String, Void, List<DeleteResult>> {
    private final DbxClientV2 mDbxClient;
    private final Callback mCallback;

    public interface Callback {
        void onDeleteComplete(List<DeleteResult> result);

        void onError(Exception e);
    }

    DeleteFileTask(DbxClientV2 dbxClient, Callback callback) {
        mDbxClient = dbxClient;
        mCallback = callback;
    }

    @Override
    protected void onPostExecute(List<DeleteResult> result) {
        super.onPostExecute(result);
        mCallback.onDeleteComplete(result);
    }

    @Override
    protected List<DeleteResult> doInBackground(String... params) {
        File remoteFolderPath = new File(Constants.DROPBOX_PATH);
        List<DeleteResult> data = new ArrayList<>();

        for (String localFile : params) {
            if (localFile != null && !localFile.isEmpty()) {
                try {
                    data.add(mDbxClient.files().deleteV2(remoteFolderPath + "/" + localFile));
                } catch (DbxException e) {
                    mCallback.onError(e);
                }
            }
        }

        return data;
    }
}

File: app/src/main/java/com/daniel/mobilepauker2/dropbox/DeleteFileTask.kt
package com.daniel.mobilepauker2.dropbox

import android.os.AsyncTask
import com.daniel.mobilepauker2.utils.Constants
import com.dropbox.core.DbxException
import com.dropbox.core.v2.DbxClientV2
import com.dropbox.core.v2.files.DeleteResult
import java.io.File
import java.util.*

/**
 * Created by Daniel on 14.03.2018.
 * Masterarbeit:
 * MobilePauker++ - Intuitiv, plattformübergreifend lernen
 * Daniel Fritsch
 * hs-augsburg
 *
 *
 * Löscht die Dateien auf Dropbox.
 */
class DeleteFileTask internal constructor(
    private val mDbxClient: DbxClientV2?,
    private val mCallback: Callback
) : AsyncTask<String, Void, List<DeleteResult>>() {

    interface Callback {
        fun onDeleteComplete(result: List<DeleteResult?>?)
        fun onError(e: Exception?)
    }

    override fun onPostExecute(result: List<DeleteResult>) {
        super.onPostExecute(result)
        mCallback.onDeleteComplete(result)
    }

    override fun doInBackground(vararg params: String): List<DeleteResult> {
        val remoteFolderPath =
            File(Constants.DROPBOX_PATH)
        val data: MutableList<DeleteResult> = ArrayList()
        for (localFile in params) {
            if (!localFile.isEmpty()) {
                try {
                    data.add(mDbxClient!!.files().deleteV2("$remoteFolderPath/$localFile"))
                } catch (e: DbxException) {
                    mCallback.onError(e)
                }
            }
        }
        return data
    }
}